compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -0,0 +1,1138 @@
+                return Optional.of(typesFactory.getTypeDescriptor(((BErrorType)
+                        ((BLangErrorConstructorExpr) bLangNode).expectedType).detailType));
+            }
+
+            return Optional.of(typesFactory.getTypeDescriptor(((BLangErrorConstructorExpr) bLangNode).expectedType));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(NamedArgumentNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        return getExpectedType(bLangNode);
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        if (!(bLangNode instanceof BLangTableConstructorExpr)) {
+            return Optional.empty();
+        }
+
+        if (node.keySpecifier().isPresent() &&
+                isWithinParenthesis(linePosition, node.keySpecifier().get().openParenToken(),
+                        node.keySpecifier().get().closeParenToken())) {
+            BType expectedType = ((BLangTableConstructorExpr) bLangNode).expectedType;
+            if (expectedType instanceof BTableType) {
+                return Optional.of(typesFactory.getTypeDescriptor(Objects.
+                        requireNonNullElse(((BTableType) expectedType).constraint, expectedType)));
+            }
+
+            if (expectedType instanceof BTypeReferenceType) {
+                BType referredType = ((BTypeReferenceType) expectedType).referredType;
+                return Optional.of(typesFactory.getTypeDescriptor(Objects.
+                        requireNonNullElse(((BTableType) referredType).constraint, expectedType)));
+            }
+
+        } else if (isWithinParenthesis(linePosition, node.openBracket(), node.closeBracket())) {
+            if (!((((BLangTableConstructorExpr) bLangNode).expectedType) instanceof BTypeReferenceType)) {
+                BType constraint = ((BTableType) ((BLangTableConstructorExpr) bLangNode).expectedType).constraint;
+                return getTypeFromBType(constraint);
+            }
+
+            BType rowType = ((BTableType) ((BTypeReferenceType) ((BLangTableConstructorExpr) bLangNode).expectedType)
+                    .referredType).constraint;
+            return Optional.of(typesFactory.getTypeDescriptor(rowType));
+        }
+
+        return getExpectedType(bLangNode);
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(RecordFieldWithDefaultValueNode node) {
+        BLangNode bLangNode = nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
+        if (bLangNode == null) {
+            return Optional.empty();
+        }
+
+        return getExpectedType(bLangNode);
```

Comment:
```suggestion
        return bLangNode == null? Optional.empty() : getExpectedType(bLangNode);
```

Wouldn't it be better if simplified like this?